index.js

```diff
@@ -195,6 +195,14 @@ function xgettext(input, options, cb) {
     if (options['files-from']) {
       input = fs.readFileSync(options['files-from'], options['from-code'])
         .split('\n')
+        .map(function (line) {
+          return line.split('').reduceRight(function(acc, c) {
```

Comment:
`split('')` has some issues. You can use `Array.from(line)` since we only support node 6 and up.